CONTRIBUTING.md

```diff
@@ -1,25 +1,75 @@
 # Overview
 
-This documents explains the processes and practices recommended for contributing enhancements to the LMA Light bundle.
+This documents explains the processes and practices recommended for
+contributing enhancements to the LMA Light bundle.
 
-- Generally, before developing enhancements to this charm, you should consider [opening an issue](https://github.com/canonical/lma-light-bundle) explaining your use case.
-- If you would like to chat with us about your use cases or proposed implementation, you can reach us on the [Charmhub Mattermost](https://chat.charmhub.io/charmhub/channels/charm-dev) or [Discourse](https://discourse.charmhub.io/).
+- Generally, before developing enhancements to this charm, you should consider
+  [opening an issue](https://github.com/canonical/lma-light-bundle) explaining
+  your use case.
+- If you would like to chat with us about your use cases or proposed
+  implementation, you can reach us on the
+  [Charmhub Mattermost](https://chat.charmhub.io/charmhub/channels/charm-dev)
+  or [Discourse](https://discourse.charmhub.io/).
 - All enhancements require review before being merged.
-  Besides the code quality and test coverage, the review will also take into  account the resulting user experience for Juju administrators using this charm.
-  Please simplify the code-review process by rebasing onto the `main` branch and avoid merge commits so we can enjoy a linear Git history!
+  Apart from code quality and test coverage, the review will also take into
+  account the Juju administrator user experience using the bundle.
 
 ## Development
 
-### Deploy from a local bundle file
+There are four charms used for development:
+- prometheus
+- alertmanager
+- grafana
+- tester
+
+### Deploy with local charms
+
+To deploy the bundle using only/some local charms you need to render the
+[`bundle-testing.yaml.j2`](tests/integration/bundle-testing.yaml.j2) template
+and then deploy the rendered bundle as usual.
+
+This can be done using tox:
 
 ```shell
-juju deploy ./bundle.yaml
+tox -e integration -- --keep-models -k test_build_and_deploy \
```

Comment:
I presume all these command line options are optional and if not provided the charm will be picked from charmhub.

Reply:
yes! but `--keep-models -k test_build_and_deploy` are key if you want to (ab)use tox :)